sklearn/decomposition/ksvd.py

```diff
@@ -0,0 +1,343 @@
+"""K-SVD dictionary learning algorithm.
+"""
+
```

Comment:
Given the structure of your code, I think it would be worth factorizing it it within `dict_learning` module, adding a new `_update_dict(method='ksvd')` method (from your `_update_atom` method and allowing for `method='ksvd' in `dictionary_learning`.